diff --git a/piff/_version.py b/piff/_version.py
index 4f4c623a..c1ba101f 100644
--- a/piff/_version.py
+++ b/piff/_version.py
@@ -12,5 +12,5 @@
 #    this list of conditions and the disclaimer given in the documentation
 #    and/or other materials provided with the distribution.
 
-__version__ = '1.3.3'
+__version__ = '1.3.3.1'
 __version_info__ = tuple(map(int, __version__.split('.')))
diff --git a/piff/pixelgrid.py b/piff/pixelgrid.py
index 4d113c77..0c10bb8c 100644
--- a/piff/pixelgrid.py
+++ b/piff/pixelgrid.py
@@ -25,6 +25,26 @@
 from .model import Model
 from .star import Star, StarData, StarFit
 
+APODIZE_PARAMS = (1.0 * 0.263, 4.25 * 0.263)
+
+
+def _ap_kern_kern(x, m, h):
+    # cumulative triweight kernel
+    y = (x - m) / h + 3
+    apval = np.zeros_like(m)
+    msk = y > 3
+    apval[msk] = 1
+    msk = (y > -3) & (~msk)
+    apval[msk] = (
+        -5 * y[msk] ** 7 / 69984
+        + 7 * y[msk] ** 5 / 2592
+        - 35 * y[msk] ** 3 / 864
+        + 35 * y[msk] / 96
+        + 1 / 2
+    )
+    return apval
+
+
 class PixelGrid(Model):
     """A PSF modeled as interpolation between a grid of points.
 
@@ -445,6 +465,21 @@ def getProfile(self, params):
         :returns: a galsim.GSObject instance
         """
         im = galsim.Image(params.reshape(self.size,self.size), scale=self.scale)
+
+        if APODIZE_PARAMS is not None:
+            xpix, ypix = im.get_pixel_centers()
+            # use_true_center = False below
+            dx = xpix - im.center.x
+            dy = ypix - im.center.y
+            r2 = dx**2 + dy**2
+
+            apwidth, aprad = APODIZE_PARAMS  # in arcsec
+            _apwidth = apwidth / self.scale  # convert to pixels
+            _aprad = aprad / self.scale  # convert to pixels
+
+            apim = im._array * _ap_kern_kern(_aprad, np.sqrt(r2), _apwidth)
+            im._array = apim / np.sum(apim) * np.sum(im._array)
+
         return galsim.InterpolatedImage(im, x_interpolant=self.interp,
                                         use_true_center=False, flux=1.)
 
diff --git a/requirements.txt b/requirements.txt
index d03b7b19..cfd6693b 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,10 +1,10 @@
-numpy>=1.17
+numpy>=1.17,<2
 scipy>=1.2
 pyyaml>=5.1
 lsstdesc.coord>=1.2
 treecorr>=4.3.1
 fitsio>=1.0
 matplotlib>=3.3
-galsim>=2.3
+galsim>=2.3,<2.5
 treegp>=0.6
 threadpoolctl>=3.1
diff --git a/tests/conftest.py b/tests/conftest.py
new file mode 100644
index 00000000..079d67ae
--- /dev/null
+++ b/tests/conftest.py
@@ -0,0 +1,4 @@
+# turn off apodization
+import piff.pixelgrid
+
+piff.pixelgrid.APODIZE_PARAMS = None
